build.sbt

```diff
@@ -116,13 +116,24 @@ lazy val otelExtension = (project in file("otel-extension"))
   .settings(
     name := "mesmer-otel-extension",
     libraryDependencies ++= {
-      openTelemetryInstrumentation ++ openTelemetryMuzzle
+      openTelemetryInstrumentation ++
+      openTelemetryMuzzle ++
+      akkaTestkit ++
+      scalatest ++ byteBuddy
     },
     assembly / test            := {},
     assembly / assemblyJarName := "mesmer-otel-extension.jar",
-    assemblyMergeStrategySettings
+    assemblyMergeStrategySettings,
+    Test / fork              := true,
+    Test / parallelExecution := true,
+    Test / testGrouping := ((Test / testGrouping).value flatMap { group =>
+      group.tests.map { test =>
+        Tests.Group(name = test.name, tests = Seq(test), runPolicy = group.runPolicy)
+      }
+    }),
+    Test / testOnly / testGrouping := (Test / testGrouping).value
   )
-  .dependsOn(core, agent)
+  .dependsOn(core % "provided->compile;test->test", agent)
```

Comment:
```suggestion
  .dependsOn(core, agent)
```

Otherwise the app fails during startup with: 

```
[error] ERROR io.opentelemetry.javaagent.OpenTelemetryAgent
[error] java.lang.NoClassDefFoundError: io/scalac/mesmer/core/module/MesmerModule
[error]         at io.scalac.mesmer.agent.config.MesmerConfigPropertySourceProvider$.modules(MesmerConfigPropertySource.scala:11)
[error]         at io.scalac.mesmer.agent.config.MesmerConfigPropertySourceProvider$.getProperties(MesmerConfigPropertySource.scala:24)
[error]         at io.scalac.mesmer.agent.config.MesmerConfigPropertySourceProvider.getProperties(MesmerConfigPropertySource.scala)
[error]         at io.scalac.mesmer.agent.config.MesmerConfigPropertySource.getProperties(MesmerConfigPropertySource.java:13)
[error]         at io.opentelemetry.javaagent.tooling.config.ConfigInitializer.loadSpiConfiguration(ConfigInitializer.java:47)
[error]         at io.opentelemetry.javaagent.tooling.config.ConfigInitializer.initialize(ConfigInitializer.java:30)
[error]         at io.opentelemetry.javaagent.tooling.AgentInstaller.<clinit>(AgentInstaller.java:81)
[error]         at io.opentelemetry.javaagent.tooling.AgentStarterImpl.start(AgentStarterImpl.java:65)
[error]         at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:37)
[error]         at io.opentelemetry.javaagent.OpenTelemetryAgent.startAgent(OpenTelemetryAgent.java:55)
[error]         at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:44)
[error]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]         at java.lang.reflect.Method.invoke(Method.java:498)
[error]         at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
[error]         at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
[error] Caused by: java.lang.ClassNotFoundException: io.scalac.mesmer.core.module.MesmerModule
[error]         at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
[error]         at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
[error]         at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
[error]         ... 17 more
```

Reply:
I'm not sure if we even need provided->compile in agent sbt config as well. [It means](https://www.scala-sbt.org/1.x/docs/Multi-Project.html#Per-configuration+classpath+dependencies) that "provided configuration in agent depends on compile configuration in core". Why would such sophisticated configuration be needed in agent that is meant only to be a fat jar added as a javaagent (not as a provided-scope dependency)? 

Anyways when we merge everything in otel-extension It will work in both variants.